### PR TITLE
Feature/zcs 11657 : Create New SOAP API for testing S3 bucket config

### PR DESCRIPTION
**Pull request by rupeshtelus**
creating a new SOAP API to test the S3 configuration which will be accepting the following set of parameters and at the server-side, it will make a test request to the s3 bucket to verify if that bucket is reachable or not.

URL
Bucket Name
Access Key
Secret Key
Region

Created new API
ValidateS3BucketReachableRequest
ValidateS3BucketReachableResponse

Request - 
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:zimbra" xmlns:urn1="urn:zimbraAdmin">
soapenv:Header
urn:context
urn:authToken0_6a08477669d3d95ceadf1c38197a64ff47ed542c_69643d33363a31393531643831632d313362372d346139662d386337352d3262306537363963636465363b6578703d31333a313635383236313934343239383b61646d696e3d313a313b747970653d363a7a696d6272613b753d313a613b7469643d393a3937303333343737363b76657273696f6e3d31353a392e312e305f424554415f343330373b637372663d313a313b</urn:authToken>
urn:csrfToken4e5ca0e2943404a0f8649dce8d604fb43cc48174</urn:csrfToken>
</urn:context>
</soapenv:Header>
soapenv:Body
<urn1:ValidateS3BucketReachableRequest
url="https://idtnzfdm3vg6.compat.objectstorage.us-ashburn-1.oraclecloud.com/"
bucketName="s3_hsmtest"
region="com.synacor.zimbra.store.location.DefaultLocationFactory"
accessKey="ba5afac8d971aa1c438744a63b76c27403ae507c"
secrateKey="7Ogvx+9ZLkTBw3eIokFtzlpLEuMQdbl2IeZjr+EoJ7k="/>
</soapenv:Body>
</soapenv:Envelope>

Response - 
<soap:Envelope xmlns:soap="[http://schemas.xmlsoap.org/soap/envelope/">](http://schemas.xmlsoap.org/soap/envelope/%22%3E)
soap:Header

</soap:Header>
soap:Body

S3 bucket connected Successfully

</soap:Body>
</soap:Envelope>

### CODE REVIEW

File: soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2012 Synacor, Inc.
```
Comment (zimsuchitgupta) [inline soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java:4]:
```suggestion
 * Copyright (C) 2022 Synacor, Inc.
```

**Review by zimsuchitgupta — CHANGES_REQUESTED** [review]

File: soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java (hunk 1 above)
Comment (rupeshtelus) [reply soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java]: changed

**Review by zimsuchitgupta — APPROVED** [review]

File: soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableResponse.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.soap.admin.type.AdminAttrsImpl;
+
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name = AdminConstants.E_VALIDATE_S3_BUCKET_REACHABLE_RESPONSE)
+public class ValidateS3BucketReachableResponse extends AdminAttrsImpl {
+
+    public ValidateS3BucketReachableResponse() {
```
Comment (dasiyogesh) [inline soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableResponse.java:31]: doesn't the response need some info about the S3 bucket is reachable or not?

File: soap/src/java/com/zimbra/soap/JaxbUtil.java [hunk 3]
```diff
@@ -1159,7 +1159,9 @@ public final class JaxbUtil {
             com.zimbra.soap.admin.message.CreateS3BucketConfigRequest.class,
             com.zimbra.soap.admin.message.CreateS3BucketConfigResponse.class,
             com.zimbra.soap.admin.message.DeleteS3BucketConfigRequest.class,
-            com.zimbra.soap.admin.message.DeleteS3BucketConfigResponse.class
+            com.zimbra.soap.admin.message.DeleteS3BucketConfigResponse.class,
+            com.zimbra.soap.admin.message.ValidateS3BucketReachableRequest.class,
```
Comment (dasiyogesh) [inline soap/src/java/com/zimbra/soap/JaxbUtil.java:1163]: there is no handler registered for handling the request, please register handler class for handling the request

**Review by dasiyogesh — CHANGES_REQUESTED** [review]
update the api response and add the missing handler

File: soap/src/java/com/zimbra/soap/JaxbUtil.java (hunk 3 above)
Comment (rupeshtelus) [reply soap/src/java/com/zimbra/soap/JaxbUtil.java:1163]: handler method is there in other repo ( zm-hsm-store )
https://github.com/Zimbra/zm-hsm-store/pull/15

File: soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    private String bucketName;
+
+    @XmlAttribute(name=GlobalExternalStoreConfigConstants.A_S3_REGION /* name */, required=false)
+    private String region;
+
+    @XmlAttribute(name=GlobalExternalStoreConfigConstants.A_S3_ACCESS_KEY /* name */, required=true)
+    private String accessKey;
+
+    @XmlAttribute(name=GlobalExternalStoreConfigConstants.A_S3_SECRET_KEY /* name */, required=true)
+    private String secrateKey;
```
Comment (RakeshAMore) [inline soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java:51]: misspelled  it should be `secretKey`

File: soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+/**
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required true
+ * @zm-api-command-description Get S3 Bucket Config
+ */
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name = AdminConstants.E_VALIDATE_S3_BUCKET_REACHABLE_REQUEST)
+public class ValidateS3BucketReachableRequest extends AdminAttrsImpl {
+
+    @XmlAttribute(name=GlobalExternalStoreConfigConstants.A_S3_URL /* name */, required=true)
```
Comment (abhishekdeshmukh2209) [inline soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java:38]: change name in /*name */ in this line.
it needs to be the field that is used.

Comment (inpati) [reply soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java:38]: Done

Comment (inpati) [reply soap/src/java/com/zimbra/soap/admin/message/ValidateS3BucketReachableRequest.java]: Done

**Review by RakeshAMore — APPROVED** [review]

**Review by vishnukantjangid — APPROVED** [review]

**Pull request closed without merging**